snapshotter/demux/snapshotter.go

```diff
@@ -0,0 +1,252 @@
+
+type DemuxSnapshotter struct {
+	cache            cache.Cache
+	fetchSnapshotter cache.SnapshotterProvider
+}
+
+func NewSnapshotter(cache cache.Cache, fetchSnapshotter cache.SnapshotterProvider) snapshots.Snapshotter {
+	return &DemuxSnapshotter{cache, fetchSnapshotter}
+}
+
+func (d *DemuxSnapshotter) Stat(ctx context.Context, key string) (snapshots.Info, error) {
+	contextLogger := log.G(ctx).WithField("function", "Stat")
+	namespace, err := getNamespaceFromContext(ctx, contextLogger)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+	logger := contextLogger.WithField("namespace", namespace)
+
+	snapshotter, err := d.getSnapshotterFromCache(ctx, namespace, logger)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+
+	info, err := snapshotter.Stat(ctx, key)
+	if err != nil {
+		contextLogger.WithError(err).Error(proxiedFunctionCallErrorString)
+		return snapshots.Info{}, err
+	}
+	return info, nil
+}
+
+func (d *DemuxSnapshotter) Update(ctx context.Context, info snapshots.Info, fieldpaths ...string) (snapshots.Info, error) {
+	contextLogger := log.G(ctx).WithField("function", "Update")
+	namespace, err := getNamespaceFromContext(ctx, contextLogger)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+	logger := contextLogger.WithField("namespace", namespace)
+
+	snapshotter, err := d.getSnapshotterFromCache(ctx, namespace, logger)
+	if err != nil {
+		return snapshots.Info{}, err
+	}
+
+	updatedInfo, err := snapshotter.Update(ctx, info, fieldpaths...)
+	if err != nil {
+		contextLogger.WithError(err).Error(proxiedFunctionCallErrorString)
+		return snapshots.Info{}, err
+	}
+	return updatedInfo, nil
+}
+
+func (d *DemuxSnapshotter) Usage(ctx context.Context, key string) (snapshots.Usage, error) {
+	contextLogger := log.G(ctx).WithField("function", "Usage")
+	namespace, err := getNamespaceFromContext(ctx, contextLogger)
+	if err != nil {
+		return snapshots.Usage{}, err
+	}
+	logger := contextLogger.WithField("namespace", namespace)
+
+	snapshotter, err := d.getSnapshotterFromCache(ctx, namespace, logger)
+	if err != nil {
+		return snapshots.Usage{}, err
+	}
+
+	usage, err := snapshotter.Usage(ctx, key)
+	if err != nil {
+		contextLogger.WithError(err).Error(proxiedFunctionCallErrorString)
+		return snapshots.Usage{}, err
+	}
+	return usage, nil
+}
+
+func (d *DemuxSnapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, error) {
+	contextLogger := log.G(ctx).WithField("function", "Mounts")
+	namespace, err := getNamespaceFromContext(ctx, contextLogger)
+	if err != nil {
+		return []mount.Mount{}, err
+	}
+	logger := contextLogger.WithField("namespace", namespace)
+
+	snapshotter, err := d.getSnapshotterFromCache(ctx, namespace, logger)
+	if err != nil {
+		return []mount.Mount{}, err
+	}
+
+	mounts, err := snapshotter.Mounts(ctx, key)
+	if err != nil {
+		contextLogger.WithError(err).Error(proxiedFunctionCallErrorString)
+		return []mount.Mount{}, err
+	}
+	return mounts, nil
```

Comment:
As @kzys mentioned, here, Prepare, and View need to iterate over the mounts and call `vm. AddLocalMountIdentifier`

Reply:
Oh, good catch. I have a rough draft of it. Just need to tighten some linter stuff + add some unit tests.